main/lsp/LSPTypechecker.cc

```diff
@@ -318,7 +318,16 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
     auto sorted = sortParsedFiles(*gs, *errorReporter, move(resolved));
     const auto presorted = true;
     const auto cancelable = false;
-    pipeline::typecheck(*gs, move(sorted), config->opts, workers, cancelable, std::nullopt, presorted);
+
+    std::vector<core::FileRef> extraFilesToFlush;
+    if (isNoopUpdateForRetypecheck) {
+        for (auto &file : updates.updatedFileIndexes) {
+            extraFilesToFlush.emplace_back(file.file);
+        }
+    }
```

Comment:
Out of curiosity, did you consider an alternative where we pass the `__package.rb` into the call to `pipeline::typecheck`, which would mean we don't need to pass this extra flag into `typecheck`?

Reply:
I'm asking because this turns `pipeline::typecheck` into a leaky abstraction. To use it properly, you have to know about LSP and how the fast/slow path works, whereas before it was a simple contract where you pass in files and it typechecks them.

Since being able to use this function correctly involves LSP-specific knowledge (fast path / slow path and the files that they contain), it would be nice to deal with that entirely in `main/lsp` instead of in `main/pipeline`.